### PR TITLE
🐛Check resolved URL for google.com during getEncryptedDocumentKey

### PR DESCRIPTION
**Pull request by elijahsoria**
--Checks if the resolvedViewerUrl contains google.com; if not, then we grab the local document key instead of google.com

### CODE REVIEW

File: extensions/amp-subscriptions/0.1/viewer-subscription-platform.js [hunk 1]
```diff
@@ -85,11 +85,11 @@ export class ViewerSubscriptionPlatform {
       'productId': this.currentProductId_,
       'origin': this.origin_,
     });
-    // TODO(chenshay): Viewer Matching: We don't know which viewer it actually
-    // is. Need to check the viewerUrl to know, or more specificlly iterate via
-    // configured platforms and check whether any of these support the viewer.
+    const viewerUrl = this.viewer_.getResolvedViewerUrl().includes('google.com')
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/viewer-subscription-platform.js]: I don't know, unfortunately, if this is currently guaranteed to exist. I think a better chance to do this is using `viewer.maybeGetMessagingOrigin`. But I'm not sure at this point what's the best way to confirm this. We may need to debug GNews.

Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/viewer-subscription-platform.js]: Also, please using `includes` from `string.js` instead.

Comment (elijahsoria) [reply extensions/amp-subscriptions/0.1/viewer-subscription-platform.js]: Done.

Comment (elijahsoria) [reply extensions/amp-subscriptions/0.1/viewer-subscription-platform.js]: Modified the code to check for a "cryptokeys" param in the URL and then look in the cryptokeys JSON object for the keys inside of the param. Currently, we fall back to google.com but I plan on removing that in the future.

File: extensions/amp-subscriptions/0.1/viewer-subscription-platform.js [hunk 2]
```diff
@@ -85,12 +85,24 @@ export class ViewerSubscriptionPlatform {
...
-    // is. Need to check the viewerUrl to know, or more specificlly iterate via
-    // configured platforms and check whether any of these support the viewer.
-    const encryptedDocumentKey = this.serviceAdapter_.getEncryptedDocumentKey(
+    // Defaulting to google.com for now.
+    // TODO(@elijahsoria): Remove this line and only rely on what is returned
+    // in the cryptokeys param.
+    let encryptedDocumentKey = this.serviceAdapter_.getEncryptedDocumentKey(
       'google.com'
     );
+    const cryptokeysNames = this.viewer_.getParam('cryptokeys');
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/viewer-subscription-platform.js]: How about we just do this: `const cryptokeyNames = viewer.getParam(...) || 'google.com'` and then do the cycle and everything else.

**Pull request merged**